src/query/server/server.go

```diff
@@ -722,36 +730,94 @@ func newStorages(
 	return fanoutStorage, cleanup, nil
 }
 
+func remoteZoneStorage(
+	remoteAddresses []string,
+	lookbackDuration time.Duration,
+	tagOptions models.TagOptions,
+	poolWrapper *pools.PoolWrapper,
+	readWorkerPool xsync.PooledWorkerPool,
+) (storage.Storage, error) {
+	if len(remoteAddresses) == 0 {
+		// No addresses; skip.
+		return nil, nil
```

Comment:
Hm, perhaps make this an error?  This would be a very strange config to call "good" and might end up with unintended mistakes in config that is hard to debug.

Reply:
So this is a "valid" case, where act as a server to remote zones but are not connected as a client (maybe a case for setting up dedicated servers to host cross zone traffic or something like that)